source/material_model/rheology/visco_plastic.cc

```diff
@@ -518,15 +569,40 @@ namespace aspect
             derivatives->viscosity_derivative_wrt_strain_rate[i] =
               Utilities::derivative_of_weighted_p_norm_average(out.viscosities[i],
                                                                volume_fractions,
-                                                               composition_viscosities,
-                                                               composition_viscosities_derivatives,
+                                                               current_isostrain_values.composition_viscosities,
+                                                               composition_viscosity_derivatives_wrt_strain_rate,
                                                                viscosity_averaging_p);
             derivatives->viscosity_derivative_wrt_pressure[i] =
               Utilities::derivative_of_weighted_p_norm_average(out.viscosities[i],
                                                                volume_fractions,
-                                                               composition_viscosities,
-                                                               composition_dviscosities_dpressure,
+                                                               current_isostrain_values.composition_viscosities,
+                                                               composition_viscosity_derivatives_wrt_pressure,
                                                                viscosity_averaging_p);
+
+            if (enable_dilation)
+              {
+                std::vector<double> dilation_values(n_compositions);
+                for (unsigned int composition_index = 0; composition_index < n_compositions; ++composition_index)
+                  dilation_values[composition_index] = current_isostrain_values.dilation_rhs_terms[composition_index] -
+                                                       current_isostrain_values.dilation_lhs_terms[composition_index] * in.pressure[i];
+
+                double dummy_averaged_parameter = numbers::signaling_nan<double>();
+
+                // always use arithmetic average for plastic dilation terms
+                derivatives->dilation_derivative_wrt_strain_rate[i] =
+                  Utilities::derivative_of_weighted_p_norm_average(dummy_averaged_parameter,
+                                                                   volume_fractions,
+                                                                   dilation_values,
+                                                                   composition_dilation_derivatives_wrt_strain_rate,
+                                                                   1);
```

Comment:
please make this more explicit:
```suggestion
                                                                   /*arithmetic average =*/ 1);
```